realm/realm-library/src/main/java/io/realm/RealmSchema.java

```diff
@@ -36,121 +33,69 @@
     // Caches Class Strings to their Schema object
-    private final Map<String, RealmObjectSchema> dynamicClassToSchema = new HashMap<String, RealmObjectSchema>();
+    private final Map<String, RealmObjectSchema> dynamicClassToSchema = new HashMap<>();
 
     private final BaseRealm realm;
-    private long nativePtr;
-    ColumnIndices columnIndices; // Cached field look up
 
     /**
      * Creates a wrapper to easily manipulate the current schema of a Realm.
      */
     RealmSchema(BaseRealm realm) {
         this.realm = realm;
-        this.nativePtr = 0;
     }
 
-    /**
-     * Creates a wrappor to easily manipulate Object Store schemas. This constructor should only be called by
-     * proxy classes during validation of schema.
-     */
-    RealmSchema() {
-        // This is the case where the schema is created from the proxy classes.
-        // dynamicClassToSchema is used to keep track of which model classes have been processed.
-        this.realm = null;
-        this.nativePtr = 0;
-        // TODO: create a Object Store realm::Schema object and store the native pointer
-    }
-
-
-    RealmSchema(ArrayList<RealmObjectSchema> realmObjectSchemas) {
-        long list[] = new long[realmObjectSchemas.size()];
-        for (int i = 0; i < realmObjectSchemas.size(); i++) {
-            list[i] = realmObjectSchemas.get(i).getNativePtr();
-        }
-        this.nativePtr = nativeCreateFromList(list);
-        this.realm = null;
-    }
-
-    public long getNativePtr() {
-        return this.nativePtr;
-    }
-
-    public void close() {
-        if (nativePtr != 0) {
-            Set<RealmObjectSchema> schemas = getAll();
-            for (RealmObjectSchema schema : schemas) {
-                schema.close();
-            }
-            nativeClose(nativePtr);
-        }
-    }
+    @Override
+    public void close() { }
 
     /**
      * Returns the Realm schema for a given class.
      *
      * @param className name of the class
      * @return schema object for that class or {@code null} if the class doesn't exists.
      */
+    @Override
     public RealmObjectSchema get(String className) {
         checkEmpty(className, EMPTY_STRING_MSG);
-        if (realm == null) {
-            if (contains(className)) {
-                return dynamicClassToSchema.get(className);
-            } else {
-                return null;
-            }
-        } else {
-            String internalClassName = TABLE_PREFIX + className;
-            if (realm.sharedRealm.hasTable(internalClassName)) {
-                Table table = realm.sharedRealm.getTable(internalClassName);
-                RealmObjectSchema.DynamicColumnMap columnIndices = new RealmObjectSchema.DynamicColumnMap(table);
-                return new RealmObjectSchema(realm, table, columnIndices);
-            } else {
-                return null;
-            }
-        }
+
```

Comment:
so the `if(realm == null) { ... } else { ... }` branches were divided into two different classes?

When could `realm == null` be true for a RealmSchema? DynamicRealm vs typed Realm?